lib/archethic/p2p/message.ex

```diff
@@ -1226,13 +1226,11 @@ defmodule ArchEthic.P2P.Message do
     }
   end
 
-  def process(%GetFirstPublicKey{address: address}) do
-    case TransactionChain.get_first_transaction(address, [:previous_public_key]) do
-      {:ok, %Transaction{previous_public_key: key}} ->
-        %FirstPublicKey{public_key: key}
-
-      {:error, :transaction_not_exists} ->
-        %NotFound{}
+  # Returns the first public_key for a given public_key and if the public_key is used for the first time, return the same public_key.
+  def process(%GetFirstPublicKey{public_key: public_key}) do
+    case TransactionChain.get_first_public_key(public_key) do
```

Comment:
I think the case statement is not required anymore, as you are expecting a single result

Reply:
so should we just directly return it as return value is always expected

Reply:
I think

Reply:
Thanks, that makes more sense to directly return